nifi-nar-bundles/nifi-parquet-bundle/nifi-parquet-processors/src/main/java/org/apache/nifi/parquet/record/ParquetRecordReader.java

```diff
@@ -61,10 +61,13 @@ public ParquetRecordReader(
         final Long offset = Optional.ofNullable(variables.get(ParquetAttribute.RECORD_OFFSET))
                 .map(Long::parseLong)
                 .orElse(null);
+        final String recordCount = variables.get(ParquetAttribute.RECORD_COUNT);
 
-        recordsToRead = Optional.ofNullable(variables.get(ParquetAttribute.RECORD_COUNT))
-                .map(Long::parseLong)
-                .orElse(null);
+        if (offset != null && recordCount != null) {
+            recordsToRead = Long.parseLong(recordCount);
+        } else {
+            recordsToRead = null;
+        }
```

Comment:
Is this necessary? Isn't `recordsToRead` already null from line 47?

Reply:
It is, because of the `final` modifier. Otherwise you get compile error.